src/unified_graphics/routes.py

```diff
@@ -18,6 +19,27 @@
 bp = Blueprint("api", __name__)
 
 
+def parse_filters(query: MultiDict) -> dict:
+    def parse_value(value):
+        if "::" in value:
+            return [parse_value(tok) for tok in value.split("::")]
+
+        if value in ["true", "false"]:
+            return value == "true"
+
+        try:
+            return float(value)
+        except ValueError:
+            return value
+
+    filters = {}
+    for col, value_list in query.lists():
+        val = [parse_value(val) for val in value_list]
+        filters[col] = val if len(val) > 1 else val[0]
+
+    return filters
```

Comment:
This function unpacks and validates the query string, is that correct? It also collapses the `MultiDict` into a `dict` so a set of filters like so:

```python
query = MultiDict([
    ('color', 'red'),
    ('size', '10'),
    ('active', 'true'),
    ('tags', 'summer::warm'),
    ('tags', 'fall::cider'),
])
```

becomes:

```python
{
    'color': 'red',
    'size': 10.0,
    'active': True,
    'tags': [['summer', 'warm'], ['fall', 'cider']]
}
```

Reply:
That’s right. I’ll add some comments…

Reply:
Awesome. Thanks! The comments you added are quite helpful.